widget/embedded/src/pages/ConfirmSwapPage.tsx

```diff
@@ -342,15 +342,13 @@ export function ConfirmSwapPage() {
           </IconButton>
         </Buttons>
       }>
-      {showWallets && (
-        <ConfirmWalletsModal
-          open={showWallets}
-          onClose={() => setShowWallets(false)}
-          onCancel={cancelFetch}
-          loading={fetchingConfirmationQuote}
-          onCheckBalance={onConfirmSwap}
-        />
-      )}
+      <ConfirmWalletsModal
```

Comment:
I'm not sure this would work as expected. On some certain situation `showWallets` will be updated (For example changing router that needs a new wallet to be connect). By removing this condition, `ConfirmWalletsModal` will not be reset anymore. I'm not sure this is expected behavior. please get a confirmation from @Ikari-Shinji-re first.

Reply:
The best idea I came up with was to reset the state in a 'useEffect' but I don't know if that is the best thing to do. Let's request @Ikari-Shinji-re for the second review and see what he thinks about that.

Reply:
I'm still not sure this is a good fix. talk with Shinji and come up with another solution please. you can make it a separate task if it takes some good amount of effort.